polkadot-protocol/protocol-components/transactions/transactions-formats.md

```diff
@@ -0,0 +1,107 @@
+
+Then, some additional data that's not part of what gets signed is required, which includes:
+
+- The spec version and the transaction version ensure that the transaction is submitted to a [compatible runtime](TODO:update-path){target=\_blank}
+- The genesis hash, which ensures that the transaction is valid for the correct chain
+- The block hash, that corresponds to the hash of the checkpoint block, enables the signature to verify that the transaction doesn't execute on the wrong fork by checking against the block number provided by the era information
+
+This process can be broken down into the following steps:
+
+1. Construct the unsigned payload
+2. Create a signing payload
+3. Sign the payload
+4. Serialize the signed payload
+5. Submit the serialized transaction
+
+An extrinsic is encoded into the following sequence of bytes just before being hex-encoded:
+
+`[ 1 ] + [ 2 ] + [ 3 ] + [ 4 ]`
+
+Where:
+
+- `[1]` contains the compact encoded length in bytes of all of the following data. Learn how compact encoding works using [SCALE](https://github.com/paritytech/parity-scale-codec){target=\_blank}
+- `[2]` is a u8 containing 1 byte to indicate whether the transaction is signed or unsigned (1 bit) and the encoded transaction version ID (7 bits)
+- `[3]` if a signature is present, this field contains an account ID, an SR25519 signature, and some extra data. If unsigned, this field contains 0 bytes
+- `[4]` is the encoded call data. This comprises 1 byte denoting the pallet to call into, 1 byte denoting the call to make in that pallet, and as many bytes as needed to encode the arguments expected by that call
+
+The metadata interface provides the way applications know how to construct a transaction correctly. An application will learn how to correctly encode a transaction using metadata types and transaction format. If a call doesn't need to be signed, then the first bit in `[2]` will be 0, so an application will know not to try decoding a signature.
+
+## Signed Extensions
+
+The Polkadot SDK provides the concept of [signed extensions](https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/reference_docs/signed_extensions/index.html){target=\_blank}, which extend an extrinsic with additional data, through the [SignedExtension](https://paritytech.github.io/try-runtime-cli/sp_runtime/traits/trait.SignedExtension.html){target=\_blank} trait.
+
+The transaction queue regularly calls signed extensions to check that a transaction is valid before it gets put in the ready queue. This is a useful safeguard for verifying that transactions won't fail in a block. Signed extensions are commonly used to enforce validation logic to protect the transaction pool from spam and replay attacks.
+
+In FRAME, a signed extension can hold any of the following types by default:
+
+- [`AccountId`](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/primitives/runtime/src/traits.rs#L1489){target=\_blank} - to encode the sender's identity
```

Comment:
Ideally, we don't link to specific lines. We gotta break out of this pattern because the files can change all the time and then we'll be referring people to the incorrect lines

Reply:
Totally agree! I didn’t notice those comments on the last PR before tackling this one, my bad! 🙏

IMO, we should set this idea as a rule of thumb for future PRs